exoscale/client.go

```diff
@@ -125,17 +125,24 @@ func (c *refreshableExoscaleClient) refreshCredentialsFromFile(path string) {
 	if err != nil {
 		fatalf("failed to read credentials file %q: %v", path, err)
 	}
+	defer f.Close()
 
-	if err = json.NewDecoder(f).Decode(&c.apiCredentials); err != nil {
-		fatalf("failed to decode credentials file %q: %v", path, err)
+	var apiCredentials exoscaleAPICredentials
+	if err = json.NewDecoder(f).Decode(&apiCredentials); err != nil {
+		infof("failed to decode credentials file %q: %v", path, err)
+		return
 	}
-	_ = f.Close()
 
-	c.exo, err = egoscale.NewClient(c.apiCredentials.APIKey, c.apiCredentials.APISecret)
+
+	client, err := egoscale.NewClient(apiCredentials.APIKey, apiCredentials.APISecret)
 	if err != nil {
-		fatalf("failed to initialize Exoscale client: %v", err)
+		infof("failed to initialize Exoscale client: %v", err)
+		return
 	}
 
+	c.exo = client
+	c.apiCredentials = apiCredentials
```

Comment:
```suggestion
	c.Lock()
	c.exo = client
	c.apiCredentials = apiCredentials
	c.Unlock()
```

The lock/unlock can probably be moved around those lines since you only mutate c here.